[docs] Remove translations section from contributing guide

nikohoffren opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

This [Crowdin](https://translate.mui.com/) link under Translations section in CONTRIBUTING.md returns Server not found. Is [this](https://crowdin.com/project/material-ui-docs) possibly the correct address?

mnajdova reviewed (commented):
We should remove this section from the Contributing guide, it's misleading, as we dropped the translations six months ago. If we decide to continue with the translations in the future, we can add it again. Would you like to update the PR to address this?

nikohoffren commented:
> We should remove this section from the Contributing guide, it's misleading, as we dropped the translations six months ago. If we decide to continue with the translations in the future, we can add it again. Would you like to update the PR to address this?

Removed Translations section from Contributing guide.
